lib/_http_server.js

```diff
@@ -351,205 +322,247 @@ function connectionListener(socket) {
     parser._consumed = true;
     parser.consume(external);
   }
-  external = null;
-  parser[kOnExecute] = onParserExecute;
+  parser[kOnExecute] =
+    onParserExecute.bind(undefined, this, socket, parser, state);
 
-  // TODO(isaacs): Move all these functions out of here
-  function socketOnError(e) {
-    // Ignore further errors
-    this.removeListener('error', socketOnError);
-    this.on('error', () => {});
+  socket._paused = false;
+}
+exports._connectionListener = connectionListener;
 
-    if (!self.emit('clientError', e, this))
-      this.destroy(e);
+function updateOutgoingData(socket, state, delta) {
+  // `outgoingData` is an approximate amount of bytes queued through all
+  // inactive responses. If more data than the high watermark is queued - we
+  // need to pause TCP socket/HTTP parser, and wait until the data will be
+  // sent to the client.
```

Comment:
I think it's probably better to keep it closer to the variable it's referring to?